pinot-common/src/main/java/org/apache/pinot/common/utils/PinotDataType.java

```diff
@@ -775,6 +969,18 @@ public static PinotDataType getPinotDataTypeForIngestion(FieldSpec fieldSpec) {
         return fieldSpec.isSingleValueField() ? PinotDataType.FLOAT : PinotDataType.FLOAT_ARRAY;
       case DOUBLE:
         return fieldSpec.isSingleValueField() ? PinotDataType.DOUBLE : PinotDataType.DOUBLE_ARRAY;
+      case BOOLEAN:
+        if (fieldSpec.isSingleValueField()) {
+          return PinotDataType.BOOLEAN;
+        } else {
+          throw new IllegalStateException("There is no multi-value type for BOOLEAN");
+        }
+      case TIMESTAMP:
+        if (fieldSpec.isSingleValueField()) {
+          return PinotDataType.TIMESTAMP;
+        } else {
+          throw new IllegalStateException("There is no multi-value type for TIMESTAMP");
```

Comment:
Same with Timestamp, would like to see multi-valued support here as well unless there is a very specific reason not to do so?

Reply:
`TIMESTAMP` is mostly used as the time field, which does not allow MV. If required, we can add the support later. Added a todo here.